src/rocketmq.erl

```diff
@@ -68,3 +69,25 @@ send_sync(Producers, Message, Timeout) when is_binary(Message) ->
 send_sync(Producers, {Message, Context}, Timeout) when is_map(Context) ->
     {_Partition, ProducerPid} = rocketmq_producers:pick_producer(Producers, Context),
     rocketmq_producer:send_sync(ProducerPid, Message, Timeout).
+
+-spec batch_send_sync(rocketmq_producers:producers(),
+                list(binary() | {binary(), rocketmq_producers:produce_context()}),
+                timer:time()) -> ok | {error, term()}.
+batch_send_sync(Producers, Messages, Timeout) ->
+    Context = fast_get_context(Messages),
+    Messages2 = normalize_batch_messages(Messages),
+    {_Partition, ProducerPid} = rocketmq_producers:pick_producer(Producers, Context),
+    rocketmq_producer:batch_send_sync(ProducerPid, Messages2, Timeout).
+
+fast_get_context([{_Bin, Context} | _]) ->
+    Context;
+fast_get_context(_) ->
+    #{}.
+
+normalize_batch_messages(Messages) ->
```

Comment:
This function should be changed after you have merged the release-v0.4.x, as I have changed the APIs in rocketmq_producer.erl to support sending messages with properties (e.g. #{key => Val, tag => Tag}).